client/common/file.c

```diff
@@ -893,6 +896,12 @@ BOOL freerdp_client_populate_settings_from_rdp_file(rdpFile* file, rdpSettings*
 		free(host);
 	}
 
+	if (~((size_t) file->GatewayAccessToken))
+	{
```

Comment:
I can't understand what is tested here ?

Reply:
@hardening this checks if `file->GatewayAccessToken != 0xFFFFFFFF` (the initial value only changed if a new value is set)
Looks strange, but is used all over the place in rdp file parser.

Reply:
That code really isn't a beauty ... but as @akallabeth mentioned it's all over the rdp parser ... maybe one day we should "fix" that ugliness in a separate PR